Makefile

```diff
@@ -76,8 +79,15 @@ update-manifests:
 
 test: test-style test-unit
 
-test-cover:
-	${DEV_ENV_CMD} test-cover.sh
+test-cover: start-test-redis
+	docker run ${DEV_ENV_OPTS} \
+		-it \
+		--link ${REDIS_CONTAINER_NAME}:TEST_REDIS \
+		${DEV_ENV_IMAGE} bash -c 'DEIS_LOGGER_REDIS_SERVICE_HOST=$$TEST_REDIS_PORT_6379_TCP_ADDR \
+		DEIS_LOGGER_REDIS_SERVICE_PORT=$$TEST_REDIS_PORT_6379_TCP_PORT \
+		test-cover.sh' \
+		|| (make stop-test-redis && false)
+	make stop-test-redis
```

Comment:
The `||` is the key. I am trying to guarantee the redis instance is stopped even if the tests fail-- which would normally fail execution of the target right then and there.

So the first time you see it is "clean up and fail."

The second time you see it, we only reach that line if tests pass and it is just "clean up."